src/plugins/intel_gna/log/log.hpp

```diff
@@ -0,0 +1,113 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <ostream>
+#include <iostream>
+
+#include "openvino/runtime/properties.hpp"
+
+/**
+ * @brief Log manager. It supports all ov::log::Level values.
+ * To use it need to set log level then just call one of the logging methods:
+ *      GnaLog::Gnalog(ov::log::Level::DEBUG);
+ *      GnaLog::LogInfo()  << "log message"; // this message will be printed
+ *      GnaLog::LogTrace() << "log message"; // this message shoudl not be printed 
+ */
+class GnaLog {
```

Comment:
I'm wondering maybe there is a sense to make this logger common? It is working with OV level property. We can have something like `ov::log`, `ov::info`, `ov::debug`, `ov::trace`

cc @ilya-lavrenov do we have something similiar in OV?

Reply:
it would be nice to have a common logging mechanism in OV. We already have the same code in AUTO plugin (https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/auto/utils/log.hpp and several other files) and VPUX (with their own implementation).

@wangleis can we combine all the implementation and propose a common OV level logging API?